src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,249 @@
+                    } catch (ArrayIndexOutOfBoundsException e) {
+                        throw new EmptyDeadlineException();
+                    }
+                    break;
+                case "event":
+                    try {
+                        addEvent(parsedInput[1]);
+                    } catch (ArrayIndexOutOfBoundsException e) {
+                        throw new EmptyEventException();
+                    }
+                    break;
+                default:
+                    throw new InvalidCommandException();
+                }
+            } catch (InvalidCommandException e) {
+                System.out.println(style(e.getMessage()));
+            }
+            saveTasks();
+            input = sc.nextLine();
+        }
+    }
+
+    private static void greet() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        String greeting = "Hello, I'm Duke!\n"
+                + "What can I do for you?";
+        System.out.println(logo + style(greeting));
+    }
+
+    private static void exit() {
+        String bye = "Goodbye. Hope to see you again soon!";
+        System.out.println(style(bye));
+        System.exit(0);
+    }
+
+    private static void addTodo(String description) {
+        Todo todo = new Todo(description);
+        tasks.add(todo);
+        printTask(todo);
+    }
+
+    private static void addDeadline(String info) throws MissingDeadlineException, InvalidDeadlineException {
+        try {
+            String[] parsedInfo = info.split("\\s*/by\\s*", 2);
+            Deadline deadline = new Deadline(parsedInfo[0], LocalDate.parse(parsedInfo[1]));
+            tasks.add(deadline);
+            printTask(deadline);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new MissingDeadlineException();
+        } catch (DateTimeParseException e) {
+            throw new InvalidDeadlineException();
+        }
+    }
+
+    private static void addEvent(String info) throws MissingTimeException {
+        try {
+            String[] parsedInfo = info.split("\\s*/at\\s*", 2);
+            Event event = new Event(parsedInfo[0], parsedInfo[1]);
+            tasks.add(event);
+            printTask(event);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new MissingTimeException();
+        }
+    }
+
+    private static void printTask(Task task) {
+        String numberOfTasks;
+        if (tasks.size() == 1) {
+            numberOfTasks = "There is now 1 task in the list.";
+        } else {
+            numberOfTasks = "There are now " + tasks.size() + " tasks in the list.";
+        }
+        System.out.println(style("Got it. I've added this task:\n  " + task + "\n" + numberOfTasks));
+    }
+
+    private static void list() {
```

Comment:
Can explore different approach to minimise arrow_head cause by if else statement.